rerun_cpp/CMakeLists.txt

```diff
@@ -97,33 +90,133 @@ if("${RERUN_C_LIB}" STREQUAL "")
     message(FATAL_ERROR "RERUN_C_LIB is not set.")
 endif()
 
-target_link_libraries(rerun_sdk PRIVATE ${RERUN_C_LIB})
+# For cleanliness, create a rerun_c target with the correct dependencies.
+# This allows other targets to depend on rerun_c directly if they need to.
+add_library(rerun_c STATIC IMPORTED)
+set_target_properties(rerun_c PROPERTIES IMPORTED_LOCATION ${RERUN_C_LIB})
+
+if(APPLE)
+    target_link_libraries(rerun_c INTERFACE "-framework CoreFoundation" "-framework IOKit")
+elseif(UNIX) # if(LINUX) # CMake 3.25
+    target_link_libraries(rerun_c INTERFACE "-lm -ldl -pthread")
+elseif(WIN32)
+    target_link_libraries(rerun_c INTERFACE ws2_32.dll Bcrypt.dll Userenv.dll ntdll.dll)
+endif()
+
+target_link_libraries(rerun_sdk PRIVATE rerun_c)
 
 # -----------------------------------------------------------------------------
 # Arrow:
-option(RERUN_ARROW_LINK_SHARED "Link to the Arrow shared library" ON)
+# Using the static arrow library by default.
+# This makes the setup a lot easier on Windows where we otherwise need to put Arrow.dll either in path or copy it with the exectuable.
+option(RERUN_ARROW_LINK_SHARED "Link to the Arrow shared library" OFF)
+option(RERUN_ARROW_EXTERNAL_PROJECT "If enabled, arrow will be added as an external project and built with the minimal set required by the Rerun C++ SDK" ON)
```

Comment:
Nit: although `ExternalProject` is the name of the utility, `RERUN_ARROW_EXTERNAL_PROJECT` sounds like the exact opposite of "Build arrow as part of the rerun project"

Something like `RERUN_BUILD_ARROW` makes more sense to me.

Reply:
or RERUN_DOWNLOAD_AND_BUILD_ARROW maybe